src/components/ReactQuery/QueryHandler.tsx

```diff
@@ -1,27 +1,51 @@
-import type { QueryStatus } from 'react-query'
-import type { ReactElement } from 'react'
+import type { ComponentProps, FC, ReactElement } from 'react'
 import ProgressSpinner from '../UI/Progress/ProgressSpinner'
+import { QueryStatus } from "react-query"
+import Skeleton from "../UI/Skeleton"
+import ConditionalAnimation from "../UI/Animation/ConditionalAnimation"
 
 
-interface QueryHandlerProps {
-	children: ReactElement;
-	status: QueryStatus;
+const loadingComponents: Record<string, FC> = {
+	spinner:  (props: LoadingComponents["spinner"]) => <ProgressSpinner {...props}/>,
+	skeleton: (props: LoadingComponents["skeleton"]) => <Skeleton {...props}/>,
 }
 
-const QueryHandler = ({ children, status }: QueryHandlerProps) => {
-	return (
-		<>
-			{status === 'loading' && (
-				<ProgressSpinner/>
-			)}
 
-			{status === 'error' && 'Error occurred while getting the data!'}
+interface LoadingComponents {
+	spinner: ComponentProps<typeof ProgressSpinner>
+	skeleton: ComponentProps<typeof Skeleton>
+}
+
+interface QueryHandlerProps<LoadingComponent extends keyof typeof loadingComponents> {
+	children: ReactElement | undefined
+	status: QueryStatus
+	loadingComponent?: LoadingComponent
+	loadingComponentProps?: LoadingComponent extends keyof LoadingComponents ? LoadingComponents[LoadingComponent] : never
+}
+
 
-			{status === 'success' && (
-				children
-			)}
+const defaultProps = {
+	loadingComponent:      'spinner',
+	loadingComponentProps: {}
+}
+
+const QueryHandler = <LoadingComponent extends keyof typeof loadingComponents = "spinner">({
+	children,
+	status,
+	loadingComponent,
+	loadingComponentProps,
+}: QueryHandlerProps<LoadingComponent> & typeof defaultProps) => {
+	const LoadingComponent = loadingComponents[loadingComponent]
+
+	return (
+		<>
```

Comment:
how about using switch case here?

Reply:
See this [blog post](https://blog.cleancoder.com/uncle-bob/2021/03/06/ifElseSwitch.html) talking about if statements vs switch case. :)

Reply:
I have read the comment about ordering the imports and found this as an opportunity to search about it on google, I have found [this article](https://dev.to/otamnitram/sorting-your-imports-correctly-in-react-213m) which's talking about it, and created a [story](https://app.shortcut.com/anki/story/37/create-a-role-for-imports-order) in Clubhouse as well.